libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1877,10 +1907,13 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 			toString(arguments.size()) +
 			"."
 		);
+		return;
 	}
-	else if (!functionType->takesArbitraryParameters() && parameterTypes.size() != arguments.size())
+
+	if (!_functionType->takesArbitraryParameters() && parameterTypes.size() != arguments.size())
```

Comment:
Starting from here, the only thing that happens is checking that the number and names of arguments is correct and that their types match, rigth? Perhaps this would be a nice candidate for extracting into a function whose names clearly states everything it does.

Reply:
I have moved argument/parameter checking into a separate method TypeChecker::typeCheckFunctionCallCompatibleArguments().